### PR TITLE
212 refactor `models` folder

### LINKED ISSUE #212: Break up and sort out imports/exports in `models` folder
- [x] exports sorted
- [x] imports sorted
- [x] file split

### PR DESCRIPTION
**Pull request by VincentPikand**
Closes #212

### CODE REVIEW

**Review by kmariuszk — CHANGES_REQUESTED** [review]
I have two questions about your decisions and thus some suggestions.

1. Why did you decide to keep the `utils` folder?
2. Since you created the `differentiable/flux` directory why you didn't create something similar for the `laplace_redux.jl` file? I had a similar discussion with Rauno about this categorization and maybe you can create another directory e.g., `differentiable/laplace_redux` or `differentiable/custom`? Thanks to that the `differentiable` folder would have a cleaner structure.

File: src/models/Models.jl [hunk 1]
```diff
@@ -3,6 +3,25 @@ module Models
...
+using Statistics
+
+include("utils.jl")
+
+include("differentiable/differentiable.jl")
+
+include("plotting/default.jl")
+include("plotting/voronoi.jl")
+
+include("pretrained/pretrained.jl")
```
[the remark this answers is not in the document]
Comment (VincentPikand) [reply src/models/Models.jl:24]: No, they are meant to be separated a bit

**Comment by VincentPikand** [comment]
> I have two questions about your decisions and thus some suggestions.
> 
>     1. Why did you decide to keep the `utils` folder?
> 
>     2. Since you created the `differentiable/flux` directory why you didn't create something similar for the `laplace_redux.jl` file? I had a similar discussion with Rauno about this categorization and maybe you can create another directory e.g., `differentiable/laplace_redux` or `differentiable/custom`? Thanks to that the `differentiable` folder would have a cleaner structure.

1) I think you mean `utils.jl`, not `utils` folder. I decided to keep it because I couldn't think of a good way to separate those functions that would be meaningful and not completely arbitrary.
2) a `custom` folder is undescriptive and if more models are created, then a more relevant structuring can be created. For now, it can stay there.

File: src/models/Models.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (kmariuszk) [reply src/models/Models.jl:24]: Why? As far as I know, we're not doing this anywhere else and from a user perspective I always skip those `using` and `include` lines when I go through code. If I needed to find some specific bits of information about imports, I'd use `ctrl+f` anyway and having all necessary imports in one place seems to be convenient enough. 

Also, please leave it to a reviewer to decide whether a conversation is resolved.

**Comment by kmariuszk** [comment]
> > I have two questions about your decisions and thus some suggestions.
> > ```
> > 1. Why did you decide to keep the `utils` folder?
> > 
> > 2. Since you created the `differentiable/flux` directory why you didn't create something similar for the `laplace_redux.jl` file? I had a similar discussion with Rauno about this categorization and maybe you can create another directory e.g., `differentiable/laplace_redux` or `differentiable/custom`? Thanks to that the `differentiable` folder would have a cleaner structure.
> > ```
> 
>     1. I think you mean `utils.jl`, not `utils` folder. I decided to keep it because I couldn't think of a good way to separate those functions that would be meaningful and not completely arbitrary.
> 
>     2. a `custom` folder is undescriptive and if more models are created, then a more relevant structuring can be created. For now, it can stay there.

Yes, I meant `utils.jl` and after going through this file once again it makes sense to leave it the way it is.

I'd still create one of those folders to keep this stuff separate, I'd argue that leaving a random `laplace_redux.jl` file could be even more confusing and creating one more folder wouldn't hurt.

File: src/models/Models.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (kmariuszk) [reply src/models/Models.jl:24]: May I ask for the link to the file from which you took a screenshot? 🙂

[the remark this answers is not in the document]
Comment (kmariuszk) [reply src/models/Models.jl:24]: Great! I'm fully convinced your approach makes sense. Thank you for the clarification.

**Review by RaunoArike — APPROVED** [review]
Seems like a straightforward PR, didn't find any mistakes or questionable decisions. Approved.

File: src/models/differentiable/differentiable.jl [hunk 2]
```diff
@@ -5,7 +5,7 @@ Base type for differentiable models.
 abstract type AbstractDifferentiableModel <: AbstractFittedModel end
 
 """
-Base type for differentiable models written in pure Julia.
+Base type for differentiable models written in pure Flux.
```
Comment (kmariuszk) [inline src/models/differentiable/differentiable.jl:8]: `Pure Flux`? I think it's supposed to be just `Flux`.

**Review by kmariuszk — CHANGES_REQUESTED** [review]
Everything looks great except that `pure Flux` thingy. Please also add an informative description of this PR.

**Review by kmariuszk — APPROVED** [review]

**Pull request merged**